enterprise/signals/utils.py

```diff
@@ -93,12 +95,27 @@ def _sample_conditional(self, params, n=1, gp=False, variance=True):
             for i, model in enumerate(self.pta.pulsarmodels):
                 for sig in model._signals:
                     if sig.signal_type in ["basis", "common basis"]:
+
                         sb = sig.get_basis(params=params)
                         nb = sb.shape[1]
 
                         if nb + ntot > len(b):
                             raise IndexError("Missing parameters! You need to set combine=False in your GPs.")
 
+                        if "timing_model" in sig.name and len(self.tm_params) > 0:
+                            if self.psr is None:
+                                raise ValueError("Need to input psr to get timing model param names")
+                            else:
+                                for tm_par in self.tm_params:
+
+                                    tm_idx = list(self.psr.fitpars).index(tm_par)
+                                    save_name = sig.name.split("_")[0] + "_" + tm_par
+
+                                    if gp:
```

Comment:
We're 8 indentations deep at this point. Can it be cleaned up at all?

Reply:
I cleaned it up a bit with the philosophy "fail fast"

Reply:
I cleaned it up a bit with the philosophy "fail fast"